lib/sequencescape_client.rb

```diff
@@ -35,44 +35,43 @@ def self.update_extraction_attributes(instance, attrs, username = 'test')
   end
 
   def self.purpose_by_name(name)
-    client.plate_purpose.all.select { |p| p.name===name }.first
+    client.plate_purpose.all.find { |p| p.name === name }
   end
 
-  def self.create_plate(purpose_name, attrs)
-    attrs = {}
+  def self.create_plate(purpose_name)
     purpose = purpose_by_name(purpose_name) || purpose_by_name('Stock Plate')
-    purpose.plates.create!(attrs)
-  end
-
-  def self.get_study_by_name(name)
-    get_study_searcher_by_name.first(name: name)
-  rescue Sequencescape::Api::ResourceNotFound => exception
-    return nil
+    purpose.plates.create!({})
   end
 
-  def self.get_study_searcher_by_name
-    @@study_searcher ||= client.search.all.select { |s| s.name == Rails.configuration.searcher_study_by_name }.first
-  end
-
-  def self.get_searcher_by_barcode
-    @@searcher ||= client.search.all.select { |s| s.name == Rails.configuration.searcher_name_by_barcode }.first
-  end
-
-  def self.get_remote_asset(barcode)
+  def self.find_by_barcode(barcode)
     find_by(barcode: barcode)
   end
 
-  def self.find_by_uuid(uuid, opts = nil)
+  def self.find_by_uuid(uuid)
     find_by(uuid: uuid)
   end
 
+  def self.labware(conditions)
```

Comment:
This covers the new import of labware. Things aren't quite as elegant as I'd like for tube racks, as their structure is quite differen.